### ISSUE TITLE
Nested TypedDict Cannot Use Enclosing Parameter for "total" kwarg

### ISSUE THREAD

**Issue by jomido** [microsoft/pyright#4042]
## What 

When a nested `TypedDict`  uses an enclosing parameter for its `total` kwarg, an erroneous type error is raised:

```python
from typing import TypedDict


def foo(a: bool):  # pylance warn: "a" is not accessed
    class Bar(TypedDict, total=a):  # pylance type error "Expected "total" parameter to have a value of True or False"
        x: int

    return Bar
```

**Expected behavior**

There should neither be a warning nor a type error.

**VS Code extension or command-line**

Pylance v2022.10.10

**Comment by jomido** [comment]
Thanks for this explanation, Eric.
